samza-core/src/main/java/org/apache/samza/clustermanager/AbstractContainerAllocator.java

```diff
@@ -35,20 +36,51 @@
 
 /**
  * {@link AbstractContainerAllocator} makes requests for physical resources to the resource manager and also runs
- * a processor on an allocated physical resource. Sub-classes should override the assignResourceRequests()
- * method to assign resource requests according to some strategy.
+ * a processor on an allocated physical resource.
  *
- * See {@link ContainerAllocator} and {@link HostAwareContainerAllocator} for two such strategies
+ * <ul>
+ *  <li>
+ *    In case of host-affinity enabled, each request ({@link SamzaResourceRequest} contains a processorId which
```

Comment:
Regardless of host-affinity enabled or not, all SamzaResourceRequests contain a processorId